README.md

```diff
@@ -10,213 +10,302 @@
-
-The SDK also tracks entry and exit of functions which increases the verbosity of the logs. This will be useful when you want to track the transitions within the codebase. To do so, you need to set log level to TRACE and add the following to the cmake file:
-`add_definitions(-DLOG_STREAMING)`
-Note: This log level is extremely VERBOSE and could flood the files if using file based logging strategy.
 
-#### Cross-Compilation
-If you wish to cross-compile `CC` and `CXX` are respected when building the library and all its dependencies. See our [ci.yml](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/develop/.github/workflows/ci.yml) for an example of this. Every commit is cross compiled to ensure that it continues to work.
-Please note that GStreamer is not cross-compiled as a part of the cross-compilation of the KVS-SDK, customers will have to cross-compile it separately.
 
+> [!NOTE]
+> For more build configuration options, see [Cmake Arguments](#cmake-arguments).
 
-#### CMake Arguments
-You can pass the following options to `cmake ..`.
+#### Compile:
+_Mac and Linux_
+```bash
+make
+```
+_Windows_
+```bat
+nmake
+```
 
-* `-DBUILD_GSTREAMER_PLUGIN` -- Build kvssink GStreamer plugin
-* `-DBUILD_JNI` -- Build C++ wrapper for JNI to expose the functionality to Java/Android
-* `-DBUILD_DEPENDENCIES` -- Build depending libraries from source
-* `-DBUILD_TEST=TRUE` -- Build unit/integration tests, may be useful for confirm support for your device. `./tst/producerTest`
-* `-DCODE_COVERAGE` --  Enable coverage reporting
-* `-DCOMPILER_WARNINGS` -- Enable all compiler warnings
-* `-DADDRESS_SANITIZER` -- Build with AddressSanitizer
-* `-DMEMORY_SANITIZER` --  Build with MemorySanitizer
-* `-DTHREAD_SANITIZER` -- Build with ThreadSanitizer
-* `-DUNDEFINED_BEHAVIOR_SANITIZER` Build with UndefinedBehaviorSanitizer
-* `-DALIGNED_MEMORY_MODEL` Build for aligned memory model only devices. Default is OFF.
-* `-DBUILD_LOG4CPLUS_HOST` Specify host-name for log4cplus for cross-compilation. Default is OFF.
-* `-DCONSTRAINED_DEVICE` Set the thread stack size to 0.5MB, needed for Alpine builds
+<br>
 
-#### To Include JNI
+### Run the Samples
+Included are sample applications for streaming and ingesting to a KVS stream. To stream to KVS, the GStreamer samples use an implemented `appsink` sink element, and the kvssink samples use the custom `kvssink` sink element.
+> [!TIP]
+> More on [GStreamer elements](https://gstreamer.freedesktop.org/documentation/application-development/basics/elements.html?gi-language=c).
 
-JNI examples are NOT built by default.  If you wish to build JNI you MUST add `-DBUILD_JNI=TRUE` when running `cmake`:
+<br>
 
+#### GStreamer appsink Samples
+To stream media from the device's camera and audio sources, create a stream in the [AWS KVS console](https://console.aws.amazon.com/kinesisvideo/home) and run the following sample:
 ```bash
-cmake -DBUILD_JNI=TRUE
+./kvs_gstreamer_audio_video_sample <your-stream-name>
```

Comment:
What happens if there's no region set here?

Reply:
DEFAULT_AWS_REGION is used. I added a note of this to the new AWS Authentication section.